tests/performance/run_server_performance_test.py

```diff
@@ -0,0 +1,346 @@
+import os
+import random
+import signal
+import subprocess
+import threading
+import time
+from collections import defaultdict
+
+
+VERBOSE = False
+FINISH = False
+PROCESSES = []
+
+
+def finish_test(signum, frame):
+    print('-----> Performance test stops. '
+          'Please wait for stopping all subprocesses. <-----')
+
+    global FINISH
+    FINISH = True
+
+    global PROCESSES
+    for proc in PROCESSES:
+        try:
+            proc.terminate()
+        except OSError:
+            pass
+
+
+signal.signal(signal.SIGINT, finish_test)
+
+
+def return_duration(func):
+    """
+    This decorator makes the applied function return its duration. The original
+    return value gets lost.
+    """
+    def func_wrapper(*args, **kwargs):
+        before = datetime.datetime.now()
+        func(*args, **kwargs)
+        after = datetime.datetime.now()
+        return (after - before).total_seconds()
+
+    return func_wrapper
+
+
+def print_process_output(message, stdout, stderr):
+    global VERBOSE
+
+    if not VERBOSE:
+        return
+
+    print(message)
+    print('-' * 20 + 'stdout' + '-' * 20)
+    print(stdout)
+    print('-' * 20 + 'stderr' + '-' * 20)
+    print(stderr)
+    print('-' * (40 + len('stdout')))
+
+
+def parse_arguments():
+    parser = argparse.ArgumentParser(
+        description='Performance tester for CodeChecker storage.',
+        epilog='This test simulates some user actions which are performed on '
+               'a CodeChecker server. The test instantiates the given number '
+               'of users. These users perform a run storage, some queries and '
+               'run deletion. The duration of all tasks is measured. These '
+               'durations are written to the output file at the end of the '
+               'test in CSV format. The tasks are performed for all report '
+               'directories by all users.')
+
+    parser.add_argument('input',
+                        type=str,
+                        metavar='file/folder',
+                        nargs='+',
+                        default='~/.codechecker/reports',
+                        help="The analysis result files and/or folders.")
+    parser.add_argument('--url',
+                        type=str,
+                        metavar='PRODUCT_URL',
+                        dest='product_url',
+                        default='localhost:8001/Default',
+                        required=True,
+                        help="The URL of the product to store the results "
+                             "for, in the format of host:port/ProductName.")
+    parser.add_argument('-o', '--output',
+                        type=str,
+                        required=True,
+                        help="Output file name for printing statistics.")
+    parser.add_argument('-u', '--users',
+                        type=int,
+                        default=1,
+                        help="Number of users")
+    parser.add_argument('-t', '--time',
```

Comment:
Wouldn't it be better to use `--timeout` instead of `--time` for full argument name? You wrote in the help that this argument is a timeout.